Core/NoOpRunner.Core/Shapes/PowerUp.cs

```diff
@@ -0,0 +1,66 @@
+﻿using NoOpRunner.Core.Entities;
+using NoOpRunner.Core.Enums;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NoOpRunner.Core.Shapes
+{
+    public class PowerUp : BaseShape
+    {
+        // Player 1 PowerUp | Player 2 PowerUp
+        Dictionary<string, string> powerUps = new Dictionary<string, string>(){
+            { "speed_boost", "rocket" },
+            { "invisibility", "proximity_mine" },
+            { "invulnerablity", "saw" },
+            { "double_jump", "knockback_bomb" }
+        };
+
+        public PowerUp(int centerPosX, int centerPosY, int[] platformXCoords, int[] platformYCoords) : base(centerPosX, centerPosY)
+        {
+            int randomLocation = RandLocation(platformXCoords, platformYCoords);
+
+            SpawnPowerUp(platformXCoords[randomLocation], platformYCoords[randomLocation] + 2);
+        }
+
+        public void SpawnPowerUp(int x, int y) // TODO (Image instead of color)
+        {
+            Color powerup = RandomPowerUp();
+
+            MapShapeX(x, y, 1, powerup);
+        }
+
+        public Color RandomPowerUp()
+        {
+            int powerUp = RandomNumber.Instance.Next(0, powerUps.Count);
+            String key = powerUps.Keys.ElementAt(powerUp);
+
+            switch (key)
+            {
+                case "speed_boost": return Color.Red;
```

Comment:
If you are using string literals more than once, they should be moved to constants in some external static class, or atleast as private variables.